src/core/dbus-execute.c

```diff
@@ -2244,6 +2245,47 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
+        } else if (streq(name, "TemporaryFileSystem")) {
+                _cleanup_strv_free_ char **l = NULL;
+                char **p;
+
+                r = sd_bus_message_read_strv(message, &l);
+                if (r < 0)
+                        return r;
+
+                STRV_FOREACH(p, l) {
+                        char *i = *p;
+
+                        if (startswith(i, "-"))
+                                i++;
+
+                        if (!path_is_absolute(i))
+                                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid %s", name);
+
+                        path_kill_slashes(i);
```

Comment:
hmm, it's bad enough we don't maintain these directory lists in fully parsed structures currently but I am very sure we should expose them fully normalized however these days, i.e. this property should be exposed as `a(bss)`, i.e. as array of triplets consisting of a boolean (for the ignore_enoent value), a path string and an option string...

The only reasons we don't expose ignore_enoent for the other options is that the boolean was added later on, and changing dbus property types after they have been introduced is nasty. But for new properties this doesn't apply, we really should expose them with full data, and fully normalized from day one.

Reply:
OK. Updated.